src/devices_models/devices/thermal_generation.jl

```diff
@@ -1203,21 +1200,35 @@ function cost_function(
                         COST_EPSILON * first_pair[2]
                     ) / (first_pair[1] * first_pair[2])
             end
-            if slopes[1] < 0 || slopes[1] <= slopes[2]
+            if slopes[1] < 0 || slopes[1] > slopes[2]
                 throw(IS.ConflictingInputsError("The PWL cost data provided for generator $(PSY.get_name(d)) is not compatible with a No Min Cost."))
             end
+            g_cost = JuMP.GenericAffExpr{Float64, _variable_type(psi_container)}()
+
+            pwlvars = JuMP.@variable(
+                psi_container.JuMPmodel,
+                [i = 1:length(cost_component)],
+                base_name = "{$(var)}_{pwl}",
+                start = 0.0,
+                lower_bound = 0.0,
+                upper_bound = PSY.get_breakpoint_upperbounds(cost_component)[i]
+            )
             for (ix, pwlvar) in enumerate(pwlvars)
-                JuMP.add_to_expression!(gen_cost, slopes[ix] * pwlvar)
-                container[(PSY.get_name(d), t, ix)] = pwlvar
+                JuMP.add_to_expression!(g_cost, slopes[ix] * pwlvar)
             end
-
             c = JuMP.@constraint(
                 psi_container.JuMPmodel,
-                variable == sum([pwlvar for (ix, pwlvar) in enumerate(pwlvars)])
+                var == sum([pwlvar for (ix, pwlvar) in enumerate(pwlvars)])
             )
-            JuMP.add_to_expression!(gen_cost, c)
+
+            if export_pwl_vars
+                for (ix, v) in enumberate(pwlvars)
```

Comment:
Typo? It seems this code needs testing. This should have been caught by a test that hits this line.